src/ImageSharp/Advanced/AotCompilerTools.cs

```diff
@@ -25,180 +40,510 @@ namespace SixLabors.ImageSharp.Advanced
     [ExcludeFromCodeCoverage]
     internal static class AotCompilerTools
     {
-        static AotCompilerTools()
-        {
-            System.Runtime.CompilerServices.Unsafe.SizeOf<long>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<short>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<float>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<double>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<byte>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<int>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<Block8x8>();
-            System.Runtime.CompilerServices.Unsafe.SizeOf<Vector4>();
-        }
-
         /// <summary>
         /// This is the method that seeds the AoT compiler.
         /// None of these seed methods needs to actually be called to seed the compiler.
         /// The calls just need to be present when the code is compiled, and each implementation will be built.
         /// </summary>
+        /// <remarks>
+        /// This method doesn't actually do anything but serves an important purpose...
+        /// If you are running ImageSharp on iOS and try to call SaveAsGif, it will throw an exception:
+        /// "Attempting to JIT compile method... OctreeFrameQuantizer.ConstructPalette... while running in aot-only mode."
+        /// The reason this happens is the SaveAsGif method makes heavy use of generics, which are too confusing for the AoT
+        /// compiler used on Xamarin.iOS. It spins up the JIT compiler to try and figure it out, but that is an illegal op on
+        /// iOS so it bombs out.
+        /// If you are getting the above error, you need to call this method, which will pre-seed the AoT compiler with the
+        /// necessary methods to complete the SaveAsGif call. That's it, otherwise you should NEVER need this method!!!
+        /// </remarks>
+        [Preserve]
         private static void SeedEverything()
```

Comment:
the only nitpick i have is, this method could be called something else like `SeedPixelImplementations` or `SeedPIxelTypes`

Reply:
Feel free to change it once we do the WebP/Tiff updates.